lib/common.sh

```
@@ -0,0 +1,316 @@
+# Validate that a symlink points to the expected location
+# Usage: validate_symlink "/path/to/link" "/expected/source"
+validate_symlink() {
+    local link="$1"
+    local expected_source="$2"
+    
+    if [ ! -L "$link" ]; then
+        log_error "Not a symlink: $link"
+        return 1
+    fi
+    
+    local actual_source=$(readlink "$link")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)